src/db_adapters/accounts.rs

```diff
@@ -152,7 +152,59 @@ pub(crate) async fn handle_accounts(
             }
         }
 
-        for value in accounts_to_insert {
+        // [Implicit accounts](https://docs.near.org/docs/roles/integrator/implicit-accounts)
+        // pretend to be created on each transfer to these accounts and cause some confusion
+        // Resolving the issue https://github.com/near/near-indexer-for-explorer/issues/68 to avoid confusion
+        // we block updating `created_by_receipt_id` for implicit accounts that were not deleted
+        // (have `deleted_by_receipt_id` NOT NULL)
+        // For this purpose we separate such accounts from others to handle them properly
+        let (implicit_accounts_to_recreate, other_accounts_to_update): (
+            Vec<models::accounts::Account>,
+            Vec<models::accounts::Account>,
+        ) = accounts_to_create_or_update.into_iter().partition(|model| {
+            model.account_id.len() == 64 && model.deleted_by_receipt_id.is_none()
+        });
+
+        for value in implicit_accounts_to_recreate {
+            let target = schema::accounts::table
+                .filter(schema::accounts::dsl::account_id.eq(value.account_id))
+                .filter(schema::accounts::dsl::deleted_by_receipt_id.is_not_null()) // this filter ensures we update only "deleted" accounts
```

Comment:
There is a rare corner case that I don't think we need to address right now since I don't see a use-case for such case, but I just want to record it:

The database state: existing non-deleted implicit account (deleted_by_receipt_id is NULL)

The receipt: `DeleteAccount` + `Transfer` (re-creating the account immediately)

Expected result: created_by_receipt_id is updated, deleted_by_receipt_id remains NULL

This implementation will lead to completely ignoring this event.

Reply:
I think this case will be ignored even earlier, where we collect all the accounts here https://github.com/near/near-indexer-for-explorer/blob/b0dbbe52f7bebf3904365805e4a3ccd90ae12c53/src/db_adapters/accounts.rs#L40-L77

Should we merge it as is?

Reply:
Yes, let's merge as is